code/modules/research/machinery/_production.dm

```diff
@@ -80,7 +80,6 @@
 		var/obj/item/I = new path(get_turf(src))
 		if(efficient_with(I.type))
 			I.material_flags |= MATERIAL_NO_EFFECTS //Find a better way to do this.
-			I.set_custom_materials(matlist)
```

Comment:
This seems like it would cause extra errors outside of what's already happening.

Are the two stacks not merging because one has mats and the other doesn't? Why do this?

Reply:
So stacks are not merging because default custom materials for cables are 10 iron 5 glass. Cables from autolathe have variable **mats_per_unit** on this default value 10/5. But everything that is being printed from protolathe override variable **mats_per_unit** by multiplying it by efficiency of protolathe in this line

Reply:
previous idea was to add check if merging is being done by cyborg, because i thought it was just for cyborg, but it also for custom material tiles or something, so i just decided to remove overriding of this value on a final product from protolathe.
sorry if my explanations are weird, not an english speaker

Reply:
Sice we are not producing grey scale tiles from protolathe, i don't see it being a problem

Reply:
for an actual fix i think we need to find other way for merging stacks, and that it shouldn't merge grey scale tiles

Reply:
> printed from protolathe override variable mats_per_unit by multiplying it by efficiency of protolathe in this line
Oh. I see. I fucking hhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhh

This might not be a solvable problem, since it's done like that to prevent mat duplication. We should just remove efficiency, but that's a bigger change

Reply:
Well if mats_per_unit check was added so grey scale tiles are not merging together, but is it needed to compare values of materials? i belive we only need to compare material, not how much material was needed. Anyway as i said actual fix would be rewriting merging code, which i don't see myself capable of without shittone of bugs

Reply:
I think the "proper" fix in this case would be making the amount of materials the protolathe puts into the produced items the constant and making how much material it consumes the variable. Ie: when a protolathe produces something at (25/50/75/100)% efficiency if the thing requires 1000 iron it consumes (4000/2000/1334/1000) iron and puts 1000 iron into the product.

Reply:
well then it would say not enough of materials, despite showing enough. says need 1000 you have 1000, but it needs actually a 4000